tardis/montecarlo/tests/test_cmontecarlo.py

```diff
@@ -1,99 +1,343 @@
+
+        'no_of_shells': 2,
+
+        'r_inner': (c_double * 2)(*[6.912e14, 8.64e14]),
+        'r_outer': (c_double * 2)(*[8.64e14, 1.0368e15]),
+
+        'time_explosion': 5.2e7,
+        'inverse_time_explosion': 1 / 5.2e7,
+
+        'electron_densities': (c_double * 2)(*[1.0e9] * 2),
+        'inverse_electron_densities': (c_double * 2)(*[1.0e-9] * 2),
+
+        'line_list_nu': (c_double * 5)(*[1.26318289e+16, 1.26318289e+16,
+                                         1.23357675e+16, 1.23357675e+16,
+                                         1.16961598e+16]),
+
+        'continuum_list_nu': (c_double * 20000)(*([1.e13] * 20000)),
+
+        'line_lists_tau_sobolevs': (c_double * 1000)(*([1.e-5] * 1000)),
+        'line_lists_j_blues': (c_double * 2)(*([1.e-10] * 2)),
+        'line_lists_j_blues_nd': 0,
+
+        'no_of_lines': 2,
+        'no_of_edges': 100,
+
+        'line_interaction_id': 0,
+        'line2macro_level_upper': (c_int64 * 2)(*([0] * 2)),
+
+        'js': (c_double * 2)(*([0.0] * 2)),
+        'nubars': (c_double * 2)(*([0.0] * 2)),
+
+        'spectrum_start_nu': 1.e14,
+        'spectrum_delta_nu': 293796608840.0,
+        'spectrum_end_nu': 6.e15,
+
+        'spectrum_virt_start_nu': 1e14,
+        'spectrum_virt_end_nu': 6e15,
+        'spectrum_virt_nu': (c_double * 20000)(*([0.0] * 20000)),
+
+        'sigma_thomson': 6.652486e-25,
+        'inverse_sigma_thomson': 1 / 6.652486e-25,
+
+        'inner_boundary_albedo': 0.0,
+        'reflective_inner_boundary': 0,
+
+        'chi_bf_tmp_partial': (c_double * 20000)(*([160.0] * 20000)),
+        't_electrons': (c_double * 2)(*([0.0] * 2)),
+
+        'l_pop': (c_double * 20000)(*([2.0] * 20000)),
+        'l_pop_r': (c_double * 20000)(*([3.0] * 20000)),
+    }
+    return StorageModel(**model_default)
+
+
+@pytest.fixture(scope="function")
+def mt_state():
+    """Fixture to return `RKState` object with default params initialized."""
+    mt_state_default = {
+        'key': (c_ulong * 624)(*([0] * 624)),
+        'pos': 0,
+        'has_gauss': 0,
+        'gauss': 0.0
+    }
+    return RKState(**mt_state_default)
 
-tests = CDLL(test_path)
 
-def test_compute_distance2boundary():
-	distance_to_boundary = 259376919351035.88
-	tests.test_compute_distance2boundary.restype = c_double
-	assert_almost_equal(tests.test_compute_distance2boundary(),
-		distance_to_boundary)
+@pytest.mark.parametrize(
+    ['packet_params', 'model_params', 'expected'],
+    [({'mu': 0.3, 'r': 7.5e14}, {'inverse_time_explosion': 1 / 5.2e7},
+      0.9998556693818854),
 
-def test_compute_distance2line():
-	distance_to_line = 7.792353908000001e+17
-	tests.test_compute_distance2line.restype = c_double
-	assert_almost_equal(tests.test_compute_distance2line(),
-		distance_to_line)
+     ({'mu': -.3, 'r': 8.1e14}, {'inverse_time_explosion': 1 / 2.6e7},
+      1.0003117541351274)]
+)
+def test_rpacket_doppler_factor(packet_params, model_params, expected, packet, model):
```

Comment:
In general I'd prefer `test_rpacket_doppler_factor(mu, r, inv_t_exp, expected):` That would make the tests more readable I think.